### ISSUE TITLE
Shape values don't match real predictions - DeepExplainer

### ISSUE THREAD

**Issue by rusmj** [shap/shap#712]
Hi,

First, thanks for this amazing project. I'm working on a text binary classifier using a Gru Neural Network with keras and now I need to explain it. I have followed your example of LSTM and DeeepExplainer but in my model, even though the base_value is correct, the output_value doesn't match the real prediction probability... so the problem is that the shap values are not correctly calculated, isn't it? Are GRU networks compatibility with DeepExplainer or I'm doing something wrong?

I add the code here:

![Captura](https://user-images.githubusercontent.com/9263879/61711335-19031000-ad54-11e9-933e-dc1c0247ca82.PNG)

Thank you for the help!!! :)

**Comment by cdpierse** [comment]
I've been having a similar issue with the force plots for the deep explainer. My model is outputting prediction probabilities like 0.99 while the force plot is only showing output values that are roughly only ever 2 above or below the explainer's base value. Would love some insight on this also.

**Comment by slundberg** [comment]
Any chance you could share a full working example of this issue I can use to debug things?

**Comment by rusmj** [comment]
Thanks for your answer @slundberg !!! I can't show my real data but I've created an other example using the same data like in your example https://slundberg.github.io/shap/notebooks/deep_explainer/Keras%20LSTM%20for%20IMDB%20Sentiment%20Classification.html

![capture1](https://user-images.githubusercontent.com/9263879/62280952-2a7fa280-b44d-11e9-9b43-1d62fe7fbb5e.PNG)

 I've created a model using a GRU neural network ...
![Capture2](https://user-images.githubusercontent.com/9263879/62281008-46834400-b44d-11e9-8595-cffc543616f4.PNG)

...and I've tried to explain the first prediction. As you can see, the base value is correct but the output value is 0.26 not the correct prediction 0.03.

![Capture3](https://user-images.githubusercontent.com/9263879/62281178-97933800-b44d-11e9-82ff-cb123ddb1e8f.PNG)

![Capture4](https://user-images.githubusercontent.com/9263879/62281186-99f59200-b44d-11e9-9d80-8b4a11df8fe1.PNG)

But if I execute the same LSTM like in your git example both base_value and output_value are correct....

![Capture5](https://user-images.githubusercontent.com/9263879/62281254-bc87ab00-b44d-11e9-899b-bb4fab5afd2e.PNG)

![Capture6](https://user-images.githubusercontent.com/9263879/62281263-be516e80-b44d-11e9-9820-9ae10fb50115.PNG)

I hope you can help me because I would like to use shap in my project :) 

Thank you again!

**Comment by dvamossy** [comment]
@slundberg I actually experimented a similar issue with DeepExplainer; and found for instance that for SELU activation function, SHAP breaks (see https://github.com/dvamossy/shap_test/blob/master/When%20DeepExplainer%20Fails.ipynb). Is it because of the linear approximations? In certain cases the SHAP values make no sense with SELU. I haven't encountered this issue with other activation functions.

**Comment by dvamossy** [comment]
Also breaks down with Leaky ReLU.

**Comment by Shivayv** [comment]
@rusmj  any update on if this got fixed or was there a solution ? I am getting similar issues too. And the magnitude of the predictions is pretty high. Any solutions or workaround ?

**Comment by Shivayv** [comment]
@slundberg Scott Can I find some articles / documentation of SHAP model convergence or accuracy ? Is that the reason why we see the values does not converge to model output ?
